server/channels/db/migrations/postgres/000108_remove_orphaned_oauth_preferences.up.sql

```diff
@@ -0,0 +1,33 @@
+DO $$
+DECLARE 
+    preferences_exist boolean := false;
+BEGIN
+    SELECT count(p.*) != 0 INTO preferences_exist FROM preferences p
+    WHERE (
+        (NOT EXISTS (
+            SELECT o.* FROM oauthaccessdata o
+            WHERE o.clientid = p.name AND o.userid = p.userid AND p.category = 'oauth_app'
+        ))
+        AND 
+        (NOT EXISTS (
+            SELECT oa.* FROM oauthauthdata oa
+            WHERE oa.clientid = p.name AND oa.userid = p.userid AND p.category = 'oauth_app'
+        ))
```

Comment:
Can you clarify this? Both look the same to me.

Reply:
`oauthaccessdata` contains the final token and refresh token, `oauthauthdata` holds the information about the code which is exchanged for the final token

Reply:
Apologies, I can't read that those are separate table :facepalm: